components/org.wso2.carbon.integrator.core/src/main/java/org/wso2/carbon/integrator/core/handler/RESTBasicAuthHandler.java

```diff
@@ -39,56 +46,122 @@
 public class RESTBasicAuthHandler implements Handler {
 
     private static final Log log = LogFactory.getLog(RESTBasicAuthHandler.class);
-
+        
+    private boolean isInitialized;
+	private String[] allowRoles;
+    
+    private Map<String, Object> properties;
+	
+	public RESTBasicAuthHandler() {
+		this.isInitialized = false;
+		this.allowRoles = null;
+        this.properties = new HashMap<String, Object>();
+	}
+	
     @Override
     public boolean handleRequest(MessageContext messageContext) {
         org.apache.axis2.context.MessageContext axis2MessageContext
                 = ((Axis2MessageContext) messageContext).getAxis2MessageContext();
         Object headers = axis2MessageContext.getProperty(
                 org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
 
-        if (headers instanceof Map) {
-            Map headersMap = (Map) headers;
-            if (headersMap.get(HTTPConstants.HEADER_AUTHORIZATION) == null) {
-                headersMap.clear();
-                axis2MessageContext.setProperty(BasicAuthConstants.HTTP_STATUS_CODE, BasicAuthConstants.SC_UNAUTHORIZED);
-                headersMap.put(BasicAuthConstants.WWW_AUTHENTICATE, BasicAuthConstants.WWW_AUTH_METHOD);
-                axis2MessageContext.setProperty(BasicAuthConstants.NO_ENTITY_BODY, true);
-                messageContext.setProperty(BasicAuthConstants.RESPONSE, BasicAuthConstants.TRUE);
-                messageContext.setTo(null);
-                Axis2Sender.sendBack(messageContext);
-                return false;
+        if (!isInitialized) {
+        	this.initialize(messageContext);
+        	isInitialized = true;
+        }
 
-            } else {
-                String credentials = (String) headersMap.get(HTTPConstants.HEADER_AUTHORIZATION);
-                if (processSecurity(credentials)) {
+        String authHeader = null;
+        if (headers != null && headers instanceof Map) {
+            Map headersMap = (Map) headers;
```

Comment:
Shall we declare this variable outside the if condition. Since it is not visible in line 83, we are getting errors.